src/aap_eda/tasks/ruleset.py

```diff
@@ -190,73 +204,92 @@ def _perform_deactivate(activation: models.Activation) -> None:
             flush=True,
         )
 
-        ActivateRulesets().deactivate(instance=instance)
+        ActivateRulesets().deactivate(
+            instance=instance, final_status=final_status
+        )
 
 
-# This is job will be scheduled by a scheduler which currently is unavailable
+# Started by the scheduler, executed by the default worker
 def monitor_activations() -> None:
-    # 1. Set status = unresponsive (all instances):
-    #    if status in [running, starting] and updated_at has timeouted
-    # 2. Restart (check latest instance):
-    #    if status is unresponsive and updated_at has timeouted
     logger.info("Task started: monitor_activations")
-    now = datetime.utcnow()
+    now = timezone.now()
+    _detect_unresponsive(now)
+    _stop_unresponsive(now)
+    _start_completed(now)
+    _start_failed(now)
+
+
+def _detect_unresponsive(now: timezone.datetime) -> None:
     running_statuses = [
         ActivationStatus.RUNNING.value,
         ActivationStatus.STARTING.value,
     ]
     cutoff_time = now - timedelta(
         seconds=int(settings.RULEBOOK_LIVENESS_TIMEOUT_SECONDS)
     )
-    models.ActivationInstance.objects.filter(
+    for instance in models.ActivationInstance.objects.filter(
         status__in=running_statuses, updated_at__lt=cutoff_time
-    ).update(status=ActivationStatus.UNRESPONSIVE.value, updated_at=now)
+    ):
+        instance.status = ActivationStatus.UNRESPONSIVE
+        instance.updated_at = now
+        save_activation_and_instance(instance, ["status", "updated_at"])
 
-    # TODO: Temporarily disable the restart logic until a proper worker can
-    # pick up the job
-    """
-    restart_policies = [
-        RestartPolicy.ALWAYS.value,
-        RestartPolicy.ON_FAILURE.value,
-    ]
+
+def _stop_unresponsive(now: timezone.datetime) -> None:
     for activation in models.Activation.objects.filter(
-        is_enabled=True, restart_policy__in=restart_policies
+        status=ActivationStatus.UNRESPONSIVE.value,
     ):
-        instance = (
-            models.ActivationInstance.objects.filter(activation=activation)
-            .order_by("-started_at")
-            .first()
+        logger.info(
+            f"Deactivate activation {activation.name} due to lost heartbeat"
+        )
+        deactivate(
+            activation_id=activation.id,
+            requester="SCHEDULER",
+            is_delete=False,
```

Comment:
Wrong parameter name, is  `delete`

Reply:
thanks. this PR has been pending for too long and experienced a lot of rebases